main/lsp/requests/completion.cc

```diff
@@ -282,6 +282,29 @@ string methodSnippet(const core::GlobalState &gs, core::SymbolRef method, core::
         fmt::format_to(result, "({})", fmt::join(typeAndArgNames, ", "));
     }
 
+    ENFORCE(!method.data(gs)->arguments().empty());
+    auto &blkArg = method.data(gs)->arguments().back();
+    ENFORCE(blkArg.flags.isBlock);
+
+    auto ctx = core::Context{gs, core::Symbols::root()};
+    auto hasBlockType = blkArg.type != nullptr && !blkArg.type->isUntyped();
```

Comment:
Will this permit the case where the block arg is defined, but given the type `T.untyped` in the signature, or will it be omitted because `T.untyped` includes `NilClass`?

Reply:
https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7Bparams(blk%3A%20T.untyped).void%7D%0A%20%20def%20self.foo(%26blk)%0A%20%20%20%20yield%20if%20block_given%3F%0A%20%20end%0Aend%0A%0AA.foo

^ because there is no error here, I don't put the snippet for the block (sorbet doesn't require it). also, we wouldn't know its arity.

I have a test for this (see `test/testdata/lsp/completion/snippet_block.*`).

Reply:
Yes, I had read through the test. It's unfortunate that we can't tell anything from that sig, but the behavior makes sense.